### ISSUE TITLE
nan loss when training with a consistent dataset using 1920-raider-waite-tarot-public-domain

### ISSUE THREAD

**Issue by samhodge-aiml** [tdrussell/diffusion-pipe#236]
I have created the following dataset

from https://huggingface.co/datasets/multimodalart/1920-raider-waite-tarot-public-domain

simply did

`git clone https://huggingface.co/datasets/multimodalart/1920-raider-waite-tarot-public-domain`

then in the directory `training` I ran the following python script

```
with open("metadata.csv","rt") as fh:
   for line in fh:
     cur_line = line.strip()
     out_file = cur_line.split(",")[0].replace(".jpg",".txt")
     text = ",".join(cur_line.split(",")[1:])
     with open(out_file,"wt") as fh2:
         fh2.write(f"{text}\n")
```

To create a single `.txt` file per `.jpg`

I adjusted the `dataset.toml`

as

```
resolutions = [768]
enable_ar_bucket = true
min_ar = 0.5
max_ar = 2.0
num_ar_buckets = 7
frame_buckets = [1, 33]

[[directory]]
path = '/projects/github/diffusion-pipe/../../datasets/1920-raider-waite-tarot-public-domain/train'

num_repeats = 1
```

and the 

training file

```
output_dir = '/projects/github/diffusion-pipe/../../datasets/1920-raider-waite-tarot-public-domain/training_run'
dataset = '/projects/github/diffusion-pipe/../../datasets/1920-raider-waite-tarot-public-domain/dataset.toml'

eval_datasets = [{name='eval', config='/projects/datasets/1920-raider-waite-tarot-public-domain/dataset-eval.toml'},]
epochs = 1000
micro_batch_size_per_gpu = 1
pipeline_stages = 1
gradient_accumulation_steps = 1
gradient_clipping = 1
warmup_steps = 20

# eval settings
eval_every_n_epochs = 20
eval_before_first_step = false
eval_micro_batch_size_per_gpu = 1
eval_gradient_accumulation_steps = 1

# misc settings
save_every_n_epochs = 5
checkpoint_every_n_minutes = 120
activation_checkpointing = 'unsloth'
partition_method = 'parameters'
save_dtype = 'bfloat16'
caching_batch_size = 1
steps_per_print = 78
video_clip_mode = 'single_beginning'
blocks_to_swap = 0

[model]
type = 'wan'
ckpt_path = '/projects/github/diffusion-pipe/../ComfyUI/models/diffusion_models/Wan2.1-T2V-1.3B'
transformer_path = '/projects/github/diffusion-pipe/../ComfyUI/models/diffusion_models/wan2.1_t2v_1.3B_fp16.safetensors'
llm_path = '/projects/github/diffusion-pipe/../ComfyUI/models/clip/umt5_xxl_fp8_e4m3fn_scaled.safetensors'
dtype = 'float16'
transformer_dtype = 'float8'
timestep_sample_method = 'logit_normal'

[adapter]
type = 'lora'
rank = 32
dtype = 'float16'

[optimizer]
type = 'AdamW8bitKahan'
lr = 2e-5
betas = [0.9, 0.99]
weight_decay = 0.01
stabilize = false
```

But all I can see if `NaN` for the loss values both in Tensorboard and in the stdout from the training script

Can you help explain this behaviour?

The model is only consuming 5.414 Gb of VRAM but GPU compute is near 100% I have a Ampere RTX a6000 GPU with 48Gb of VRAM.

**Comment by samhodge-aiml** [comment]
Of note when I previously produced the labels with Florence within ComfyUI, the loss sat between 0.10 and 0.20 but the LoRA had no effect in ComfyUI inference, when using terms that were present in the labels.

The new labels have a unique term `trtcrd`

**Comment by samhodge-aiml** [comment]
Image resolution is 1150x1920

**Comment by samhodge-aiml** [comment]
this may have been the cause

the following update to the python script strips the quotes

```
with open("metadata.csv","rt") as fh:
   for line in fh:
     cur_line = line.strip()
     out_file = cur_line.split(",")[0].replace(".jpg",".txt")
     text = ",".join(cur_line.split(",")[1:])[1:-1]
     with open(out_file,"wt") as fh2:
         fh2.write(f"{text}\n")
```

**Comment by samhodge-aiml** [comment]
I am going to try again with 500 warm up steps

**Comment by samhodge-aiml** [comment]
Looks like I was using some incorrect models from this tutorial:

https://learn.thinkdiffusion.com/depth-in-every-frame-how-wan-2-1-elevates-video-generation-using-control-loras/

**Comment by adamdavidcole** [comment]
For any folks who find themselves in the same situation: I downloaded the full model from huggingface and am only using a checkpoint path `ckpt_path = '../Wan2.1-I2V-14B-480P'` instead of pointing to my ComfyUI files. I believe the issue was with the text_encoder, which was returning NaN values (not sure why). 

After that change, I deleted my cached latents and reran the training with real loss values 👍

**Comment by jakehemmerle** [comment]
+1 choosing a non-fp8 T5 and deleting my dataset caches fixed my issue
